### PR TITLE
DEV: Convert some files to autoloading and various improvements

### PR DESCRIPTION
**Pull request by OsamaSayegh**

### CODE REVIEW

**Comment by OsamaSayegh** [comment]
Plugin spec failure will be resolved in https://github.com/discourse/discourse-data-explorer/pull/289.

**Review by ZogStriP — APPROVED** [review]

File: plugins/automation/plugin.rb [hunk 1]
```diff
@@ -39,21 +39,9 @@ def self.get_active_automation
...
   %w[
-    app/jobs/regular/discourse_automation_call_zapier_webhook
-    app/jobs/regular/discourse_automation_trigger
-    app/jobs/scheduled/discourse_automation_tracker
-    app/jobs/scheduled/stalled_topic_tracker
-    app/jobs/scheduled/stalled_wiki_tracker
-    app/queries/stalled_topic_finder
-    app/services/discourse_automation/user_badge_granted_handler
-    lib/discourse_automation/event_handlers
-    lib/discourse_automation/post_extension
```
Comment (jjaffeux) [inline plugins/automation/plugin.rb:56]: Doing this means the scheduled jobs won't be autoloaded in dev anymore, look what we did in chat (and other places) recently:

https://github.com/discourse/discourse/commit/e949684fc54e629035b8883dabd9ef2916251da1

Comment (jjaffeux) [reply plugins/automation/plugin.rb:56]: Or even better:

https://github.com/discourse/discourse/pull/26853/files

Comment (OsamaSayegh) [reply plugins/automation/plugin.rb:56]: done!

**Pull request merged**